src: kernel_config_manager: fix `kw --save | -s` behavior when `git commit` fails

davidbtadokoro opened the pull request:
After this patch, failing `kw --save | -s` because of error in commiting save operation of .config file no longer throws wrong error (undefined function) neither consolidates the operation.

Closes: #696

Commit:
src: kernel_config_manager: replace undefined function call

When using 'kw --save | -s' to add a new .config file to kw management,
if the 'git commit' command fails, the undefined function 'fail()' is
called, generating a 'fail: command not found' error. By replacing this
call for a 'complain()' call, the command not found error is not longer
thrown and the right error is printed.

WIP #696

Signed-off-by: David Tadokoro <davidbtadokoro@usp.br>

Commit:
src: kernel_config_manager: fix hanging .config files

When the 'kw --save | -s' fails because the commit in the 'kw/configs'
repo could not be made, the .config file and its metadata are added to
the management of kw anyways. By hard resetting the 'kw/config' repo,
the lingering files are removed and the repo stays consistent with the
error thrown to the user.

Closes #696

Signed-off-by: David Tadokoro <davidbtadokoro@usp.br>

rodrigosiqueira commented on src/kernel_config_manager.sh:
```diff
@@ -131,7 +131,7 @@ function save_config_file()
     warning "Warning: $name: there's nothing new in this file"
   elif [[ "$ret" -gt 0 ]]; then
-    fail "Could not save user config files"
+    complain "Could not save user config files"
```
Nice catch! Since we are here, could you replace those double quotes with single quotes?

rodrigosiqueira commented on src/kernel_config_manager.sh:
```diff
@@ -131,6 +131,7 @@ function save_config_file()
   if [[ "$ret" == 1 ]]; then
     warning "Warning: $name: there's nothing new in this file"
   elif [[ "$ret" -gt 0 ]]; then
+    git reset --hard # Aborts save operation
```
If you are fixing a bug, we need a test that reproduces the issue. However, see my comment on the issue that you created; maybe we can take a different path here.

Commit:
src: kernel_config_manager: replace double quotes with single quotes

Corrects quoting in case of expansion not needed.

WIP #696

Signed-off-by: David Tadokoro <davidbtadokoro@usp.br>

Commit:
src: kernel_config_manager: remove undesirable output

When a commit in the kw .config file manager fails, the abortion of the
save operation is made with a git reset --hard. However, the output of
this command is printed in the user stdout. With this commit, this
output is supressed as it makes no sense to the user.

WIP #696

Signed-off-by: David Tadokoro <davidbtadokoro@usp.br>

Commit:
tests: kernel_config_manager: add test case for failed git commit

This commit adds a case to contemplate the issue #696 bug.

Closes #696

Signed-off-by: David Tadokoro <davidbtadokoro@usp.br>

Commit:
tests: kernel_config_manager: corrects code style

The previous commit added code that did not follow the project's coding
style (my fault as I did not configured the pre-commit in my machine)
and this commit corrects this.

Signed-off-by: David Tadokoro <davidbtadokoro@usp.br>

rodrigosiqueira commented:
Hi @davidbtadokoro , could you reduce the total of commits to just 3?
* Commit to replacing fail to complain
* One commit for the cosmetic changes
* One commit for the fix

davidbtadokoro commented:
For sure @rodrigosiqueira 
Thought of squashing some of those commits, but for the sake of organization I'm going to close this PR and create a clone of it.
